pkg/controller/argocd/sso.go

```diff
@@ -16,15 +16,63 @@ package argocd
 
 import (
 	"context"
+	"fmt"
 
 	argoprojv1a1 "github.com/argoproj-labs/argocd-operator/pkg/apis/argoproj/v1alpha1"
 	"github.com/argoproj-labs/argocd-operator/pkg/controller/argoutil"
+	oappsv1 "github.com/openshift/api/apps/v1"
 	template "github.com/openshift/api/template/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
-var templateAPIFound = false
+const (
+	// SuccessResonse is returned when a realm is created in keycloak.
+	successResonse = "201 Created"
+	// ExpectedReplicas is used to identify the keycloak running status.
+	expectedReplicas int32 = 1
+	// ServingCertSecretName is a secret that holds the service certificate.
+	servingCertSecretName = "sso-x509-https-secret"
```

Comment:
Please search "sso-x509-https-secret" in keycloak.go and replace them with this constants.

Reply:
ack , thank you :)